pkg/api/scylla/v1/cluster_types.go

```diff
@@ -205,13 +205,11 @@ type Network struct {
 }
 
 func (s Network) GetDNSPolicy() corev1.DNSPolicy {
-	if s.DNSPolicy == "" {
-		if s.HostNetworking {
-			return corev1.DNSClusterFirstWithHostNet
-		}
-		return corev1.DNSDefault
+	if len(s.DNSPolicy) != 0 {
+		return s.DNSPolicy
 	}
-	return s.DNSPolicy
+
+	return corev1.DNSClusterFirstWithHostNet
```

Comment:
shouldn't we set DNSClusterFirstWithHostNet for hostNetworking setups and DNSClusterFirst for others?

Reply:
not based on
https://github.com/scylladb/scylla-operator/blob/6c9209d48b551558872b9d0587767672c658d5f0/vendor/k8s.io/api/core/v1/types.go#L2578-L2586
and trying it out

Reply:
but let me check kubelet code to be 100% sure

Reply:
https://github.com/kubernetes/kubernetes/blob/7735cf7f2674627283db081d61a2c45f4b70ee8f/pkg/kubelet/network/dns/dns.go#L275-L285